components/abstractions/src/main/java/com/microsoft/kiota/MultipartBody.java

```diff
@@ -0,0 +1,171 @@
+package com.microsoft.kiota;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.microsoft.kiota.serialization.Parsable;
+import com.microsoft.kiota.serialization.ParseNode;
+import com.microsoft.kiota.serialization.SerializationWriter;
+import com.microsoft.kiota.serialization.SerializationWriterFactory;
+
+/**
+ * Represents a multipart body for a request or a response.
+ */
+public class MultipartBody implements Parsable {
+	@Nonnull
+	private final String boundary = UUID.randomUUID().toString().replace("-", "");
+	/** @return the boundary string for the multipart body. */
+	@Nonnull
+	public String getBoundary() {
+		return boundary;
+	}
+	/**
+	 * The request adapter to use for the multipart body serialization.
+	 */
+	@Nullable
+	public RequestAdapter requestAdapter;
+	/**
+	 * Adds or replaces a part in the multipart body.
+	 * @param <T> the type of the part to add or replace.
+	 * @param name the name of the part to add or replace.
+	 * @param contentType the content type of the part to add or replace.
+	 * @param value the value of the part to add or replace.
+	 */
+	public <T> void addOrReplacePart(@Nonnull final String name, @Nonnull final String contentType, @Nonnull final T value) {
+		Objects.requireNonNull(name);
+		Objects.requireNonNull(contentType);
+		Objects.requireNonNull(value);
+		if (contentType.isBlank() || contentType.isEmpty())
+			throw new IllegalArgumentException("contentType cannot be blank or empty");
+		if (name.isBlank() || name.isEmpty())
+			throw new IllegalArgumentException("name cannot be blank or empty");
```

Comment:
Might be a good idea to leverage googles String library and use
Strings.IsNullOrEmpty(name) and consolidate the illegalArgumentExceptions with the requireNonNull exceptions.

Reply:
Excellent point! updated.